packages/design-system-react-native/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,171 @@
+import { useTailwind } from '@metamask/design-system-twrnc-preset';
+import React, {
+  forwardRef,
+  useState,
+  useRef,
+  useImperativeHandle,
+  useCallback,
+  useEffect,
+} from 'react';
+import { Pressable, Animated, Easing } from 'react-native';
+import type { StyleProp, ViewStyle } from 'react-native';
+
+import { Icon, IconName, IconColor, IconSize } from '../Icon';
+import { TextOrChildren } from '../temp-components/TextOrChildren';
+
+import type { CheckboxProps } from './Checkbox.types';
+
+const AnimatedView = Animated.View;
+
+export const Checkbox = forwardRef<{ toggle: () => void }, CheckboxProps>(
+  (
+    {
+      isSelected,
+      defaultIsSelected = false,
+      isDisabled = false,
+      isInvalid = false,
+      label = '',
+      labelProps,
+      onChange,
+      checkboxContainerProps,
+      checkedIconProps,
+      twClassName = '',
+      style,
+      ...props
+    },
+    ref,
+  ) => {
+    // Internal state for uncontrolled
+    const [internalSelected, setInternalSelected] = useState(defaultIsSelected);
+    const isControlled = isSelected !== undefined;
+    const currentSelected = isControlled
+      ? Boolean(isSelected)
+      : internalSelected;
+
+    // Animation values
+    const scaleAnim = useRef(new Animated.Value(1)).current;
+    const iconAnim = useRef(
+      new Animated.Value(currentSelected ? 1 : 0),
+    ).current;
+
+    // Sync icon opacity whenever selection changes
+    useEffect(() => {
+      Animated.timing(iconAnim, {
+        toValue: currentSelected ? 1 : 0,
+        duration: 300,
+        easing: Easing.ease,
+        useNativeDriver: true,
+      }).start();
+    }, [currentSelected, iconAnim]);
+
+    // Bounce effect when toggling
+    const animateScale = useCallback(() => {
+      Animated.sequence([
+        Animated.timing(scaleAnim, {
+          toValue: 1.15,
+          duration: 100,
+          easing: Easing.ease,
+          useNativeDriver: true,
+        }),
+        Animated.timing(scaleAnim, {
+          toValue: 1,
+          duration: 100,
+          easing: Easing.ease,
+          useNativeDriver: true,
+        }),
+      ]).start();
+    }, [scaleAnim]);
+
+    // Press handler: update state, notify parent, then bounce
+    const handlePress = () => {
+      if (isDisabled) {
+        return;
+      }
+      const next = !currentSelected;
+      if (!isControlled) {
+        setInternalSelected(next);
+      }
+      onChange?.(next);
+      animateScale();
+    };
+
+    useImperativeHandle(ref, () => ({ toggle: handlePress }), [handlePress]);
+
+    const tw = useTailwind();
+    const twContainerClassNames = `flex-row items-center ${
+      isDisabled ? 'opacity-50' : 'opacity-100'
+    } ${twClassName}`;
```

Comment:
This is more of a developer experience suggestion but you may have noticed that tailwind intellisense doesn't work here. However if you were to move the `tw` here it would

```suggestion
    const twContainerClassNames = tw`flex-row items-center ${
      isDisabled ? 'opacity-50' : 'opacity-100'
    } ${twClassName}`;
```